src/components/new-request/search.vue

```diff
@@ -566,16 +611,61 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin) {
     this.business = business
     this.entity_type_cd = this.business?.legalType || null
     this.setCorpNum(business?.identifier || null)
+
+    if (this.isRestoration) {
+      // Check if not XPRO and BC restorable
+      if (!this.isSelectedXproAndRestorable && this.isBcRestorable) {
+        this.setLocation(Location.BC)
+        this.setEntityTypeCd(this.corpTypeToEntityType(this.business.legalType as unknown as CorpTypeCd))
+      } else if (this.isSelectedXproAndRestorable) { // Check if XPRO and restorable
+        this.setLocation(Location.CA)
+        this.setEntityTypeCd(this.business.legalType)
+      } else {
+        this.setEntityTypeCd(null)
+      }
+    }
+  }
+
+  // Text field rules for the business lookup
+  get rules (): any {
+    return [
+      val => {
+        if (this.isRestoration) {
+          return this.isRestorable || 'This business cannot be restored.'
+        }
+        return true
+      }]
+  }
+
+  /** Returns whether the selected XPRO is restorable. */
+  get isSelectedXproAndRestorable (): boolean {
+    return XproMapping.REH.includes(this.business?.legalType)
+  }
+
+  /** Returns whether the selected business' legal type is BC and restorable. */
+  get isBcRestorable (): boolean {
+    return BcMapping.REH.includes(this.corpTypeToEntityType(this.business?.legalType as unknown as CorpTypeCd))
+  }
+
+  /** Returns whether company is restorable. */
+  get isRestorable (): boolean {
+    return this.isSelectedXproAndRestorable || this.isBcRestorable
   }
 
   /**
    * If user is authenticated, create draft business and redirect to Dashboard.
+   * If restoration/reinstatement selected, go to business dashboard.
    * If user is not authenticated, redirect to login screen then redirect back.
    */
   async incorporateNowClicked () {
     const legalType = this.entityTypeAlternateCode(this.entity_type_cd)
     if (this.getIsAuthenticated) {
-      await this.incorporateNow(legalType)
+      if (!this.isRestoration && !this.business) {
+        await this.incorporateNow(legalType)
```

Comment:
I think it would be easier to understand positive logic here, ie, in which cases to we go to business dashboard? Or, in which cases do we incorporate now?

Also, why are you checking "and not business"?

Reply:
Looks good.